### PR TITLE
modbus config parser: csv to json

### PR DESCRIPTION
**Pull request by wastrauss**
Some generated errors/warnings may show as a result of unknown config definitions. 

Run script with the following command:
`config_parser.rb flexe trion ipt flex_hybrid`

JSON output is found in imqs-timeseries-modbus-type.json

### CODE REVIEW

**Comment by JacoBriers** [comment]
Move the files under the Devices folder to Confluence rather than committing to repo.

**Comment by JacoBriers** [comment]
Add a README file with link to instructions to convert from original source files input for your script

File: config_parser/config_parser.rb [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+			end
+			build_hash(@config_file)
+			puts ""
+		end
+	end
+end
+#main
+puts "config_parser.rb help for available configs"
+#Clear JSON config file
+File.open("imqs-timeseries-modbus-type.json", "w") {|file| file.truncate(0) }
```
Comment (JacoBriers) [inline config_parser/config_parser.rb]: Make all input and output file names command line parameters to script

**Comment by JacoBriers** [comment]
Discussed with @wesseloosthuizen and think we'll remove the `group` concept in the `values` member of `ControllerTypes`. Rather duplicate the `value` and introduce a `tags` member which will contain a key-value pairs (similar to what is stored in OpenTSDB)

File: config_parser/config_parser.rb [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+@Special_type_hash = Hash.new
+
+def header_list(file)
+	CSV.foreach(file) do |row|
+		@header_list = row
+		return row
+	end
+end
+
+def check_headers
```
Comment (JacoBriers) [inline config_parser/config_parser.rb]: Good that we check this (thumbsup)

**Comment by JacoBriers** [comment]
We'll need to add a MSB/LSB `member` to JSON output to indicate byte-order for metrics

**Comment by JacoBriers** [comment]
If any value is `null` -> no need to output the JSON member

**Comment by JacoBriers** [comment]
@davidt-imqs which ruby gem do we use to pretty-print our JSON?

File: config_parser/config_parser.rb (hunk 2 above)
Comment (davidteren) [inline config_parser/config_parser.rb]: You set the array in line above correctly. Do the same for the hash
`@Special_type_hash = {}`

**Comment by davidteren** [comment]
@JacoBriers we use the following function that's part of the standard Ruby JSON lib.
a copy & paste example;
```
require 'json'
my_json = { :array => [1, 2, 3, { :sample => "hash"} ], :foo => "bar" }
puts JSON.pretty_generate(my_json)
```

Which gets you:
```
{
  "array": [
    1,
    2,
    3,
    {
      "sample": "hash"
    }
  ],
  "foo": "bar"
}
```

**Pull request merged**